src-cpp/rdkafkacpp.h

```diff
@@ -1411,6 +1449,15 @@ class RD_EXPORT KafkaConsumer : public virtual Handle {
    */
   virtual ErrorCode position (std::vector<TopicPartition*> &partitions) = 0;
 
+        /**
+         * @brief Retrieve queue for a given partition.
+         *
+         * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success in which case \p queue
+         *          is set to the fetch queue for the given partition. Else,
+         *          return an error code.
+         */
+        virtual ErrorCode partitionQueue (TopicPartition *partition,
+                                          Queue *queue) = 0;
```

Comment:
Queues are dynamically allocated objects, so I think this method should return a new queue.

Also, use snake case (get_partition_queue) rather than camel case